### DISCUSSION TITLE
Compiled osu! editor feedback (long list)

### DISCUSSION THREAD

**Discussion by NiceAesth** [ppy/osu#24161] [Ideas]
This is a copy of the google doc containing feedback from the mapping community.

Issues

- [ ] No bookmarks displayed - (https://github.com/ppy/osu/issues/20883)
- [ ] No way to add bookmarks - (https://github.com/ppy/osu/issues/20883)
- [ ] Cannot disable countdown (https://github.com/ppy/osu/issues/20883)
- [ ] Cannot add new combo to spinners - (https://github.com/ppy/osu/issues/24156)
- [x] Timeline settings don’t save (waveform/ticks/bpm) -> fixed by https://github.com/ppy/osu/pull/28505
- [x] Right click doesn’t finish sliders outside of playfield (enter does) (https://github.com/ppy/osu/pull/24181)
- [ ] Beatmap colors should always be enabled in editor (https://github.com/ppy/osu/pull/24182)
- [ ] Hit markers are too bright, hard to see stuff (a bit opinionated)
- [x] CTRL+H / CTRL+J does not prompt save when exiting the beatmap - (https://github.com/ppy/osu/issues/24065)

Missing features

- [ ] Add tooltips to options such as distance snap which show the hotkeys to change them
- [x] Translucent editor UI (for both timelines) - would massively improve visibility
- [x] CTRL+R for rotate popup (https://github.com/ppy/osu/issues/12045)
- [x] Similar keybind for scaling by fixed values (https://github.com/ppy/osu/pull/28309)
- [ ] Objects do not get deleted when placing another object on top
- [ ] Add ability to shift click on slider nodes to cycle type (https://github.com/ppy/osu/pull/28509)
- [ ] Add ability to cycle slider node type while placing the slider (https://github.com/ppy/osu/pull/28509)
- [x] Hotkey for preserving aspect ratio on scale (e.g. shift) (https://github.com/ppy/osu/pull/28309)
- [ ] Option to always use default skin in editor (or editor skin)
- [ ] Add ability to set separate skin for playtest mode
- [ ] Add ability to add difficulty from .osu file (for GDs) - or the proposed feature of mounting the beatmap folder
- [x] Hitsound additions

Miscellaneous

- [ ] Ability to hide hitsounds from timeline (for new editor design)
- [ ] Ability to quickly hide all panels for previewing the map
- [ ] Reduce size of editor elements by default (Multiple mappers have said that they use 0.80x scaling and that it still is not enough. I have resorted to using windowed mode and using a square aspect ratio to have a larger playfield)
- [ ] Bottom timeline is way too big especially if it will always be visible
- [ ] Slider nodes can very often go underneath the rotation controls on the selection box making them impossible to highlight

**Comment by bdach** [comment]
> CTRL+H / CTRL+J does not prompt save when exiting the beatmap

I can't reproduce this. It should have been fixed by https://github.com/ppy/osu/issues/24065.

**Reply by NiceAesth** [reply]
I have tested this again on master and it has indeed been fixed.

**Comment by NiceAesth** [comment]
There was another issue which caused the slider head to become the slider ball when scrolling through the map seemingly at random. ~~Can't seem to reproduce that one on master anymore either.~~ Nevermind, this is still an issue and I have found reproduction steps. It can also happen when dragging a slider on top of another slider.

https://github.com/ppy/osu/assets/8437201/ad7b6358-073f-4209-bee4-8cbcc78a71cf


Seems to already be tracked in https://github.com/ppy/osu/issues/10909

**Comment by bdach** [comment]
> Right click doesn’t finish sliders outside of playfield (enter does)

I'm not sure which is correct based on this point alone. I'd say *not* finishing sliders outside of playfield is correct, and as such, enter key is wrong? It shouldn't be possible to map offscreen sliders in the editor.

**Reply by cihe13375** [reply]
Please note that outside of playfield (goes beyond [0,512]x[0,384]) is different from offscreen. Slider tails may go beyond the [0,512]x[0,384] region, while still being fully visible under 4:3 and this is rankable.

**Reply by peppy** [reply]
Fixed in #24181.

**Comment by peppy** [comment]
> Beatmap colors should always be enabled in editor

What about skins and hitsounds?

**Reply by LeNitrous** [reply]
Under "Missing Features"
> Option to always use default skin in editor (or editor skin)

But it is more preferred if the editor has its own skin setting different from gameplay.
I suppose the order of priority would be (from most to least):

- Beatmap Skin (Toggleable)
- Editor Skin
- Gameplay Skin
- Default

**Reply by NiceAesth** [reply]
Yes, beatmap hitsounds should also take priority in the editor (or at least a toggle for this)

(+ what lenitrous said about skins)
